module/zfs/dnode_sync.c

```diff
@@ -384,12 +384,7 @@ dnode_sync_free_range_impl(dnode_t *dn, uint64_t blkid, uint64_t nblks,
 		}
 	}
 
-	/*
-	 * Do not truncate the maxblkid if we are performing a raw
-	 * receive. The raw receive sets the mablkid manually and
-	 * must not be overriden.
-	 */
-	if (trunc && !dn->dn_objset->os_raw_receive) {
+	if (trunc) {
```

Comment:
Why do we need both this code and dn_next_maxblkid to be able to reduce the maxblkid (in the raw recv case)?  Or is this just code cleanup?

Reply:
This bit is just a code cleanup. This if statement used to just say `if (trunc)`, but it was changed to help handle the maxblkid a while back. That fix was incomplete (as we are seeing here) and so this simply puts the old code back in place.